test/snippet/range/view/detail_adaptor_def_persist.cpp

```diff
@@ -0,0 +1,41 @@
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::detail
+{
+
+class persist_fn_snippet : public adaptor_base<persist_fn>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = adaptor_base<persist_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief       For ranges that are viewable, delegate to std::view::all.
+     * \returns     An instance of std::view::all.
+     */
+    template <std::ranges::ViewableRange urng_t>
+    static auto impl(urng_t && urange)
+    {
+        return std::view::all(std::forward<urng_t>(urange));
+    }
+
+    /*!\brief       For ranges that are not views and not lvalue-references, call view_persist's constructor.
+     * \returns     An instance of seqan3::detail::view_persist.
+     */
+    template <std::ranges::Range urng_t>
+    static auto impl(urng_t && urange)
+    {
+        static_assert(!std::is_lvalue_reference_v<urng_t>, "BUG: lvalue-reference in persist_fn::impl().");
+        return view_persist{std::move(urange)};
+    }
+};
+
+} // namespace seqan3::detail
```

Comment:
What happens here? Where is this snippet used?

Reply:
I have that it's used in range/view/detail.hpp

Reply:
I am against copy'n'pasting from the library code into the snippets. It will just get out of sync and it's more difficult to understand on its own. Also the snippet is only used to annotate ::detail stuff, right?

Can you undo this change and just keep referring to the snippet from the library code? I think it affects two places...

Reply:
Just for reference why I think ::detail get different treatment:

  * detail code documentation is only read by developers, i.e. people who wnat to change the library or develop library-like functionality.
  * for them the snippets demonstrate were they can find good examples inside the library. This context is valuable because it contains more clues on how to implement something.
  * If we extract the code, we remove the context which is less helpful.

Reply:
Hmmmm okay sounds good!